api/src/main/java/edu/cornell/mannlib/vitro/webapp/rdfservice/filter/AcceptableLanguages.java

```diff
@@ -25,9 +28,17 @@ public AcceptableLanguages(List<String> rawLanguageStrs) {
         log.debug("Raw language strings:" + rawLanguageStrs);
         for (String lang : rawLanguageStrs) {
             this.add(lang);
-            String baseLang = lang.split("-")[0];
-            if (!lang.equals(baseLang) && !this.contains(baseLang)) {
-                this.add(baseLang);
+            String[] subtags = lang.split(SEPARATOR);
+            int length = subtags.length;
+            for (int i = 1; i < length; i++) {
+                int lastIndex = length - i;
+                if (PRIVATE_USE_SUBTAG.equals(subtags[lastIndex - 1])) {
+                    continue;
+                }
+                String baseLang = String.join(SEPARATOR, Arrays.copyOfRange(subtags, 0, lastIndex));
+                if (!lang.equals(baseLang) && !this.contains(baseLang)) {
+                    this.add(baseLang);
+                }
```

Comment:
I believe I need this code for this [PR](https://github.com/vivo-project/Vitro/pull/329). However, I am wondering can we centralized working with languages/locales, at the moment we are struggling with acceptable languages and format of a language tag at a couple of places:
- https://github.com/vivo-project/Vitro/blob/9230bb3f23ffbfd58347ae61bb7ce55bd0d006d3/api/src/main/java/edu/cornell/mannlib/vitro/webapp/i18n/I18n.java#L314
- and https://github.com/vivo-project/Vitro/blob/9230bb3f23ffbfd58347ae61bb7ce55bd0d006d3/api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/LocaleUtility.java#L15

Reply:
also, please notice that if we define this language tag in the runtime.properties
`fr_CA_x_uqam`
once when it is loaded in a Locale object, the code 
`locale.toLanguageTag().toString()` 
will produce
`fr_CA_x_uqam`
while the code
`locale.toString()`
will produce
`fr_CA_#x-uqam`
I believe this might introduce some issues, for instance here - https://github.com/vivo-project/Vitro/blob/9230bb3f23ffbfd58347ae61bb7ce55bd0d006d3/api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/searchengine/SearchQueryUtils.java#L211:L217